src/mito2/src/region/opener.rs

```diff
@@ -427,11 +427,11 @@ pub(crate) async fn replay_memtable<S: LogStore>(
                 .unwrap_or(0);
             region_write_ctx.push_mutation(mutation.op_type, mutation.rows, OptionOutputTx::none());
         }
-    }
 
-    // set next_entry_id and write to memtable.
-    region_write_ctx.set_next_entry_id(last_entry_id + 1);
-    region_write_ctx.write_memtable();
+        // set next_entry_id and write to memtable.
```

Comment:
We should also bring the instantialization of `RegionWriteContext` (line 401) from outside the loop to inside (line 420).